src/main/java/com/github/xetorthio/jedisque/BinaryJedisque.java

```diff
@@ -0,0 +1,62 @@
+package com.github.xetorthio.jedisque;
+
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+import redis.clients.util.SafeEncoder;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+
+public class BinaryJedisque extends redis.clients.jedis.Connection {
+	static final private int DISQUE_PORT = 7711;
+	private final List<URI> uris = new ArrayList<URI>();
+	private Random randomGenerator = new Random();
+
+	public BinaryJedisque() {
+		try {
+			uris.add(new URI("disque://localhost:" + DISQUE_PORT));
+		} catch (URISyntaxException e) {
+		}
+	}
+
+	public BinaryJedisque(URI... uris) {
+		this.uris.addAll(Arrays.asList(uris));
+	}
+
+	@Override
+	public void connect() {
+		while (!this.isConnected()) {
```

Comment:
I agree... This can be as simple as just retrying the node, or can be more
complex and we could remove it for some time and then re-add it, etc.
So I am not sure which is the strategy we should use. Anyway, this can be
treated as an issue on it's own since it is not related directly yo
BinaryJedisque :)


On Mon, Jun 1, 2015 at 11:07 AM Marco Zapletal <notifications@github.com>
wrote:

> In src/main/java/com/github/xetorthio/jedisque/BinaryJedisque.java
> <https://github.com/xetorthio/jedisque/pull/5#discussion_r31428003>:
>
> > +	private Random randomGenerator = new Random();
> > +
> > +	public BinaryJedisque() {
> > +		try {
> > +			uris.add(new URI("disque://localhost:" + DISQUE_PORT));
> > +		} catch (URISyntaxException e) {
> > +		}
> > +	}
> > +
> > +	public BinaryJedisque(URI... uris) {
> > +		this.uris.addAll(Arrays.asList(uris));
> > +	}
> > +
> > +	@Override
> > +	public void connect() {
> > +		while (!this.isConnected()) {
>
> Yes. A failed node may get back to a running state (e.g., by restarting
> the same instance, assigning the hostname of the failing node to a new
> instance, etc.). Hence, retrying failed nodes allow the
> application/jedisque to get back to regular behavior without restarting the
> application.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xetorthio/jedisque/pull/5/files#r31428003>.
>